### DISCUSSION TITLE
Want to know the role of  name: 'counter',

### DISCUSSION THREAD

**Discussion by saurabhkushwaha438** [reduxjs/redux-toolkit#4440] [General]
`import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  value: 0,
}

export const counterSlice = createSlice({
  name: 'counter',
  initialState,
  reducers: {
    increment: (state) => {
      // Redux Toolkit allows us to write "mutating" logic in reducers. It
      // doesn't actually mutate the state because it uses the Immer library,
      // which detects changes to a "draft state" and produces a brand new
      // immutable state based off those changes
      state.value += 1
    },
    decrement: (state) => {
      state.value -= 1
    },
    incrementByAmount: (state, action) => {
      state.value += action.payload
    },
  },
})

// Action creators are generated for each case reducer function
export const { increment, decrement, incrementByAmount } = counterSlice.actions

export default counterSlice.reducer`

i dont know where it is used  name: 'counter', and whats its purpose

**Comment by markerikson** [comment]
@saurabhkushwaha438 it's used as the prefix for action type strings, like `"counter/increment"`.

**Reply by EskiMojo14** [reply]
it's also the default for `reducerPath`, which is used in a few places (`combineSlices` and `.selectors`)

**Reply by saurabhkushwaha438** [reply]
what if i dont use it , does action work , if yes then why we use  this [name: 'counter']extra syntax

**Reply by EskiMojo14** [reply]
no, name is a required field.
